zenodo/modules/github/cli.py

```diff
@@ -0,0 +1,307 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Click command-line interface for GitHub management."""
+
+from __future__ import absolute_import, print_function
+
+import click
+from flask.cli import with_appcontext
+from invenio_accounts.models import User
+from invenio_db import db
+from sqlalchemy.orm.exc import NoResultFound
+
+from invenio_github.api import GitHubAPI
+from invenio_github.models import Repository
+
+
+@click.group()
+def github():
+    """Management commands for GitHub."""
+
+
+def user_param(ctx, param, value):
+    """Resolve user from CLI input parameter (email or User ID)."""
+    if value is None:
+        return
+    if '@' in value:  # Try to resolve as email
+        try:
+            user = User.query.filter_by(email=value.strip()).one()
```

Comment:
Could it be kind of dangerous to use the email for identifying the user in this case? It might be that the email on Zenodo is not the same as the one used on GitHub, and this could lead to transferring/assigning repository ownership to the wrong person. Maybe always displaying a warning or doing an extra check for the user's email on GitHub (through `invenio_github.api.GitHubAPI.me`) might minimize these sort of risk or at least ring a bell.